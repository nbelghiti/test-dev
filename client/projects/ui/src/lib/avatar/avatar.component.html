<span
	class="c-avatar"
	[ngClass]="{
		'is-compact': compact
	}"
>
	<abbr [title]="name">{{initials}}</abbr>
</span>
