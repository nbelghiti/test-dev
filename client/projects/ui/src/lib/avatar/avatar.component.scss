// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size-base: 4.4rem;
$avatar-size-compact: 3.2rem;

// =============================================================================
// :: Avatar
// =============================================================================
.c-avatar {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $avatar-size-base;
	min-width: $avatar-size-base;
	height: $avatar-size-base;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-align: center;

	border-radius: 50%;
	background-color: $background-color-grey-96;

	cursor: default;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-avatar {

	&.is-compact {
		width: $avatar-size-compact;
		min-width: $avatar-size-compact;
		height: $avatar-size-compact;

		font-size: $font-size-base;
	}
}
