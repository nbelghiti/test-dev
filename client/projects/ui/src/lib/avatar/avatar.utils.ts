export const getInitials = (name: string): string => {
	const names = name.split(' ');
	let initials = names[0].slice(0, 1);

	if (names.length > 1) {
		initials += names[names.length - 1].slice(0, 1);
	}

	return initials.toUpperCase();
};
