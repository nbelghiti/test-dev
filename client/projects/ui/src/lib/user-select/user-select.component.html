<div
	uiFlyout
	class="c-user-select"
>
	<div class="c-user-select__trigger">
		<ui-user
			*ngIf="selectedUser; else emptyState"
			[user]="selectedUser"
			uiFlyoutToggle
			class="c-user-select__trigger-user"
		>
		</ui-user>

		<ui-icon-button
			*ngIf="selectedUser?.name"
			[type]="ButtonType.tertiary"
			(clicked)="removeSelectedUser(selectedUser)"
			class="u-margin-left-1x u-margin-right-1x"
		>
			<ui-icon [name]="IconName.times"></ui-icon>
		</ui-icon-button>

		<ng-template #emptyState>
			<div class="c-user-select__trigger-avatar"></div>

			<input
				uiFlyoutFocus
				class="c-user-select__trigger-input"
				placeholder="Select {{ label | translate }}"
				[formControl]="searchQuery"
			>
		</ng-template>
	</div>

	<div
		uiFlyoutContent
		class="c-flyout__content--left"
	>
		<ul
			uiFlyoutToggle
			*ngIf="users && users.length > 0; else noUsers"
		>
			<li
				*ngFor="let user of users;"
				class="c-user-select__content-item"
				(click)="clickUser(user)"
			>
				<ui-user
					[user]="user"
				></ui-user>
			</li>
		</ul>

		<ng-template #noUsers>
			<ui-empty-state
				icon="users"
				size="small"
				[message]="'NO_USERS' | translate"
			>
			</ui-empty-state>
		</ng-template>
	</div>
</div>
