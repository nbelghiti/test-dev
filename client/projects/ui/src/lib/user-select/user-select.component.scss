// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size: 3.2rem;

// =============================================================================
// :: Root
// =============================================================================
:host {
	flex-grow: 1;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-user-select__trigger {
	position: relative;

	display: flex;
	justify-content: space-between;
	align-items: center;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	border: $border-width-base dashed $border-color-grey-86;
	background-color: $white;

	cursor: pointer;
}

.c-user-select__trigger-avatar {
	position: absolute;
	top: 50%;
	left: 1rem;

	flex-shrink: 0;
	width: $avatar-size;
	height: $avatar-size;

	border-radius: 50%;

	background-color: $background-color-grey-96;

	transform: translateY(-50%);

	pointer-events: none;
}

.c-user-select__trigger-input {
	flex-grow: 1;
	padding: 1.2rem 1rem 1.2rem 5rem;

	font-size: $font-size-2x-small;
	font-weight: $font-weight-normal;
	line-height: $line-height-base;

	color: $text-color-grey-46;

	cursor: text;
}

.c-user-select__trigger-user {
	flex-grow: 1;
	padding: 1.2rem 1rem;

	font-size: $font-size-2x-small;
	font-weight: $font-weight-normal;
	line-height: $line-height-base;

	color: $text-color-grey-46;

	cursor: text;
}

.c-user-select__content-item {
	padding: 0.6rem 2rem;

	cursor: pointer;

	&:hover {
		background-color: $background-color-grey-96;
	}
}

// =============================================================================
// :: States
// =============================================================================
.c-user-select {

	&.is-open {

		.c-user-select__trigger-arrow {
			transform: rotate(180deg);
		}
	}
}
