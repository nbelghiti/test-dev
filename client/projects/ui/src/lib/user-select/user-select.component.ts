import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { FormControl } from '@angular/forms';
import * as _debounce from 'lodash.debounce';

import { ButtonType } from '../button/button.types';
import { IconName, IconSize } from '../icon/icon.types';
import { IUser } from '../user/user.types';

const debounce = _debounce;

@Component({
	selector: 'ui-user-select',
	templateUrl: './user-select.component.html',
	styleUrls: ['./user-select.component.scss'],
})

export class UserSelectComponent implements OnInit {
	@Input() public users: IUser[] = [];
	@Input() public role: string = '';
	@Input() public selectedUser: IUser = null;
	@Output() public selectUser: EventEmitter<IUser> = new EventEmitter<IUser>();
	@Output() public searchUsers: EventEmitter<string> = new EventEmitter<string>();
	@Output() public removeUser: EventEmitter<IUser> = new EventEmitter<IUser>();

	public IconName: typeof IconName = IconName;
	public IconSize: typeof IconSize = IconSize;
	public ButtonType: typeof ButtonType = ButtonType;
	public searchQuery: FormControl = new FormControl('');
	public label: string = '';

	public ngOnInit(): void {
		this.onChange();
	}

	public onChange(): void {
		this.label = `TEAM_MEMBER_ROLE__${this.role}`;
		const debouncedChange = debounce((val) => this.searchUsers.emit(val), 400);

		this.searchQuery.valueChanges.subscribe(val => {
			debouncedChange(val);
		});
	}

	public clickUser(user: IUser): void {
		this.selectUser.emit(user);
		this.searchUsers.emit('');

		this.selectedUser = {
			...user,
			role: this.role,
		};
	}

	public removeSelectedUser(user: IUser): void {
		this.removeUser.emit(user);
		this.searchUsers.emit('');

		this.selectedUser = null;
	}
}
