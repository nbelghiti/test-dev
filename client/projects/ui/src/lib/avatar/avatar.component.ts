import { Component, Input } from '@angular/core';

import { getInitials } from './avatar.utils';

@Component({
	selector: 'ui-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})

export class AvatarComponent {
	@Input() public compact: boolean = false;

	public initials = '';

	@Input() set name(name: string) {
		if (name) {
			this.initials = getInitials(name);
		}
	}
}
