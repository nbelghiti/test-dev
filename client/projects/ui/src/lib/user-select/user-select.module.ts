import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonModule } from '../button/button.module';
import { EmptyStateModule } from '../empty-state/empty-state.module';
import { FlyoutModule } from '../flyout/flyout.module';
import { IconModule } from '../icon/icon.module';
import { UserModule } from '../user/user.module';

import { UserSelectComponent } from './user-select.component';

@NgModule({
	imports: [
		CommonModule,
		UserModule,
		IconModule,
		ButtonModule,
		EmptyStateModule,
		FlyoutModule,
		ReactiveFormsModule,
		TranslateModule
	],
	declarations: [
		UserSelectComponent,
	],
	exports: [
		UserSelectComponent,
	],
})
export class UserSelectModule {}
