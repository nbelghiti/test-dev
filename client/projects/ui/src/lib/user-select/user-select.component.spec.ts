import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { AvatarComponent } from '../avatar/avatar.component';
import { IconButtonComponent } from '../button/icon-button/icon-button.component';
import { EmptyStateComponent } from '../empty-state/empty-state.component';
import { FlyoutModule } from '../flyout/flyout.module';
import { IconComponent } from '../icon/icon.component';
import { UserComponent } from '../user/user.component';

import { UserSelectComponent } from './user-select.component';

const mockUser = {
	name: 'John',
	role: '',
	type: 'engineer',
	myself: false,
};

const mockUsers = [
	mockUser,
	mockUser,
	mockUser,
];

describe('UserSelectComponent', () => {
	let component: UserSelectComponent;
	let fixture: ComponentFixture<UserSelectComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				FlyoutModule,
				ReactiveFormsModule,
				TranslateModule.forRoot(),
				RouterTestingModule
			],
			declarations: [
				UserSelectComponent,
				AvatarComponent,
				IconComponent,
				IconButtonComponent,
				UserComponent,
				EmptyStateComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(UserSelectComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});

	describe('init', () => {
		describe('without selected user', () => {
			beforeEach(() => {
				fixture = TestBed.createComponent(UserSelectComponent);
				component = fixture.componentInstance;

				fixture.detectChanges();
			});

			it('should create', () => {
				expect(component).toBeDefined();
			});

			it('should render', () => {
				expect(fixture).toMatchSnapshot();
			});

			it('should render an empty state with an input field when no selected user is provided', () => {
				expect(
					fixture.debugElement.query(By.css('input'))
				).toBeTruthy();
			});

			it('should not render an icon button to remove the user when no selected user is provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-icon-button'))
				).toBeFalsy();
			});
		});

		describe('with selected user', () => {
			beforeEach(() => {
				fixture = TestBed.createComponent(UserSelectComponent);
				component = fixture.componentInstance;
				component.selectedUser = mockUser;

				fixture.detectChanges();
			});

			it('should create', () => {
				expect(component).toBeDefined();
			});

			it('should render', () => {
				expect(fixture).toMatchSnapshot();
			});

			it('should not render an empty state with an input field when selected user is provided', () => {
				expect(
					fixture.debugElement.query(By.css('input'))
				).toBeFalsy();
			});

			it('should render an icon button to remove the user when selected user is provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-icon-button'))
				).toBeTruthy();
			});
		});

		describe('without users', () => {
			beforeEach(() => {
				fixture = TestBed.createComponent(UserSelectComponent);
				component = fixture.componentInstance;

				fixture.detectChanges();
			});

			it('should create', () => {
				expect(component).toBeDefined();
			});

			it('should render', () => {
				expect(fixture).toMatchSnapshot();
			});

			it('should render an empty state component when no users are provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-empty-state'))
				).toBeTruthy();
			});

			it('should not render a ui-user component when no users are provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-user'))
				).toBeFalsy();
			});
		});

		describe('with users', () => {
			beforeEach(() => {
				fixture = TestBed.createComponent(UserSelectComponent);
				component = fixture.componentInstance;
				component.users = mockUsers;

				fixture.detectChanges();
			});

			it('should create', () => {
				expect(component).toBeDefined();
			});

			it('should render', () => {
				expect(fixture).toMatchSnapshot();
			});

			it('should not render an empty state component when users are provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-empty-state'))
				).toBeFalsy();
			});

			it('should render a ui-user component when users are provided', () => {
				expect(
					fixture.debugElement.query(By.css('ui-user'))
				).toBeTruthy();
			});
		});
	});

	it('Should call the onChange method correctly', () => {
		expect(component.searchQuery.value).toEqual('');

		component.searchQuery.setValue('new value');

		expect(component.searchQuery.value).toEqual('new value');
	});

	it('Should call the clickUser method correctly', () => {
		expect(component.selectedUser).toEqual(null);

		component.clickUser(mockUser);

		expect(component.selectedUser).toEqual(mockUser);
	});
});
