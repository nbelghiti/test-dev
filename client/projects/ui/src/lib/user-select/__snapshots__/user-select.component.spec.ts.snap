// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelectComponent Should render correctly 1`] = `
<ui-user-select
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  label={[Function String]}
  removeUser={[Function EventEmitter]}
  role=""
  searchQuery={[Function FormControl]}
  searchUsers={[Function EventEmitter]}
  selectUser={[Function EventEmitter]}
  selectedUser="null"
  users={[Function Array]}
>
  <div
    class="c-user-select c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-user-select__trigger"
    >
      
      <div
        class="c-user-select__trigger-avatar"
      />
      <input
        class="c-user-select__trigger-input c-flyout__focus ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        placeholder="Select TEAM_MEMBER_ROLE__"
        uiflyoutfocus=""
      />
      
      
    </div>
    <div
      class="c-flyout__content--left c-flyout__content"
      ng-reflect-class="c-flyout__content--left"
      uiflyoutcontent=""
    >
      
      <ui-empty-state
        icon="users"
        ng-reflect-icon="users"
        ng-reflect-message="NO_USERS"
        ng-reflect-size="small"
        size="small"
      >
        <div
          class="c-empty-state c-empty-state--small"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-users"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            NO_USERS
          </p>
          <div
            ng-reflect-ng-class="[object Object]"
          />
        </div>
      </ui-empty-state>
      
    </div>
  </div>
</ui-user-select>
`;

exports[`UserSelectComponent init with selected user should render 1`] = `
<ui-user-select
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  label={[Function String]}
  removeUser={[Function EventEmitter]}
  role=""
  searchQuery={[Function FormControl]}
  searchUsers={[Function EventEmitter]}
  selectUser={[Function EventEmitter]}
  selectedUser={[Function Object]}
  users={[Function Array]}
>
  <div
    class="c-user-select c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-user-select__trigger"
    >
      
      <ui-user
        class="c-user-select__trigger-user c-flyout__toggle"
        ng-reflect-user="[object Object]"
        uiflyouttoggle=""
      >
        
        
        <button
          class="c-user"
          ng-reflect-klass="c-user"
          ng-reflect-ng-class="[object Object]"
          title="John"
        >
          
          <ui-avatar
            ng-reflect-compact="true"
            ng-reflect-name="John"
          >
            <span
              class="c-avatar is-compact"
              ng-reflect-klass="c-avatar"
              ng-reflect-ng-class="[object Object]"
            >
              <abbr
                title="undefined"
              >
                J
              </abbr>
            </span>
          </ui-avatar>
          <div
            class="c-user__content"
          >
            <span
              class="c-user__name"
            >
               John
              
            </span>
            <span
              class="c-user__role"
            >
               TEAM_MEMBER_ROLE__engineer 
            </span>
          </div>
        </button>
        
        
      </ui-user>
      
      <ui-icon-button
        class="u-margin-left-1x u-margin-right-1x"
        ng-reflect-type="c-button--tertiary"
      >
        
        <button
          class="c-icon-button c-icon-button--tertiary c-icon-button--border"
          ng-reflect-klass="c-icon-button"
          ng-reflect-ng-class="[object Object]"
          title=""
        >
          
          <ui-icon
            class="c-icon"
            ng-reflect-name="times"
          >
            <i
              class="c-icon__glyph   fas fa-times"
            />
          </ui-icon>
          
        </button>
        
        
      </ui-icon-button>
      
    </div>
    <div
      class="c-flyout__content--left c-flyout__content"
      ng-reflect-class="c-flyout__content--left"
      uiflyoutcontent=""
    >
      
      <ui-empty-state
        icon="users"
        ng-reflect-icon="users"
        ng-reflect-message="NO_USERS"
        ng-reflect-size="small"
        size="small"
      >
        <div
          class="c-empty-state c-empty-state--small"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-users"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            NO_USERS
          </p>
          <div
            ng-reflect-ng-class="[object Object]"
          />
        </div>
      </ui-empty-state>
      
    </div>
  </div>
</ui-user-select>
`;

exports[`UserSelectComponent init with users should render 1`] = `
<ui-user-select
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  label={[Function String]}
  removeUser={[Function EventEmitter]}
  role=""
  searchQuery={[Function FormControl]}
  searchUsers={[Function EventEmitter]}
  selectUser={[Function EventEmitter]}
  selectedUser="null"
  users={[Function Array]}
>
  <div
    class="c-user-select c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-user-select__trigger"
    >
      
      <div
        class="c-user-select__trigger-avatar"
      />
      <input
        class="c-user-select__trigger-input c-flyout__focus ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        placeholder="Select TEAM_MEMBER_ROLE__"
        uiflyoutfocus=""
      />
      
      
    </div>
    <div
      class="c-flyout__content--left c-flyout__content"
      ng-reflect-class="c-flyout__content--left"
      uiflyoutcontent=""
    >
      
      <ul
        class="c-flyout__toggle"
        uiflyouttoggle=""
      >
        
        <li
          class="c-user-select__content-item"
        >
          <ui-user
            ng-reflect-user="[object Object]"
          >
            
            
            <button
              class="c-user"
              ng-reflect-klass="c-user"
              ng-reflect-ng-class="[object Object]"
              title="John"
            >
              
              <ui-avatar
                ng-reflect-compact="true"
                ng-reflect-name="John"
              >
                <span
                  class="c-avatar is-compact"
                  ng-reflect-klass="c-avatar"
                  ng-reflect-ng-class="[object Object]"
                >
                  <abbr
                    title="undefined"
                  >
                    J
                  </abbr>
                </span>
              </ui-avatar>
              <div
                class="c-user__content"
              >
                <span
                  class="c-user__name"
                >
                   John
                  
                </span>
                <span
                  class="c-user__role"
                >
                   TEAM_MEMBER_ROLE__engineer 
                </span>
              </div>
            </button>
            
            
          </ui-user>
        </li>
        <li
          class="c-user-select__content-item"
        >
          <ui-user
            ng-reflect-user="[object Object]"
          >
            
            
            <button
              class="c-user"
              ng-reflect-klass="c-user"
              ng-reflect-ng-class="[object Object]"
              title="John"
            >
              
              <ui-avatar
                ng-reflect-compact="true"
                ng-reflect-name="John"
              >
                <span
                  class="c-avatar is-compact"
                  ng-reflect-klass="c-avatar"
                  ng-reflect-ng-class="[object Object]"
                >
                  <abbr
                    title="undefined"
                  >
                    J
                  </abbr>
                </span>
              </ui-avatar>
              <div
                class="c-user__content"
              >
                <span
                  class="c-user__name"
                >
                   John
                  
                </span>
                <span
                  class="c-user__role"
                >
                   TEAM_MEMBER_ROLE__engineer 
                </span>
              </div>
            </button>
            
            
          </ui-user>
        </li>
        <li
          class="c-user-select__content-item"
        >
          <ui-user
            ng-reflect-user="[object Object]"
          >
            
            
            <button
              class="c-user"
              ng-reflect-klass="c-user"
              ng-reflect-ng-class="[object Object]"
              title="John"
            >
              
              <ui-avatar
                ng-reflect-compact="true"
                ng-reflect-name="John"
              >
                <span
                  class="c-avatar is-compact"
                  ng-reflect-klass="c-avatar"
                  ng-reflect-ng-class="[object Object]"
                >
                  <abbr
                    title="undefined"
                  >
                    J
                  </abbr>
                </span>
              </ui-avatar>
              <div
                class="c-user__content"
              >
                <span
                  class="c-user__name"
                >
                   John
                  
                </span>
                <span
                  class="c-user__role"
                >
                   TEAM_MEMBER_ROLE__engineer 
                </span>
              </div>
            </button>
            
            
          </ui-user>
        </li>
      </ul>
      
    </div>
  </div>
</ui-user-select>
`;

exports[`UserSelectComponent init without selected user should render 1`] = `
<ui-user-select
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  label={[Function String]}
  removeUser={[Function EventEmitter]}
  role=""
  searchQuery={[Function FormControl]}
  searchUsers={[Function EventEmitter]}
  selectUser={[Function EventEmitter]}
  selectedUser="null"
  users={[Function Array]}
>
  <div
    class="c-user-select c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-user-select__trigger"
    >
      
      <div
        class="c-user-select__trigger-avatar"
      />
      <input
        class="c-user-select__trigger-input c-flyout__focus ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        placeholder="Select TEAM_MEMBER_ROLE__"
        uiflyoutfocus=""
      />
      
      
    </div>
    <div
      class="c-flyout__content--left c-flyout__content"
      ng-reflect-class="c-flyout__content--left"
      uiflyoutcontent=""
    >
      
      <ui-empty-state
        icon="users"
        ng-reflect-icon="users"
        ng-reflect-message="NO_USERS"
        ng-reflect-size="small"
        size="small"
      >
        <div
          class="c-empty-state c-empty-state--small"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-users"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            NO_USERS
          </p>
          <div
            ng-reflect-ng-class="[object Object]"
          />
        </div>
      </ui-empty-state>
      
    </div>
  </div>
</ui-user-select>
`;

exports[`UserSelectComponent init without users should render 1`] = `
<ui-user-select
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  label={[Function String]}
  removeUser={[Function EventEmitter]}
  role=""
  searchQuery={[Function FormControl]}
  searchUsers={[Function EventEmitter]}
  selectUser={[Function EventEmitter]}
  selectedUser="null"
  users={[Function Array]}
>
  <div
    class="c-user-select c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-user-select__trigger"
    >
      
      <div
        class="c-user-select__trigger-avatar"
      />
      <input
        class="c-user-select__trigger-input c-flyout__focus ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        placeholder="Select TEAM_MEMBER_ROLE__"
        uiflyoutfocus=""
      />
      
      
    </div>
    <div
      class="c-flyout__content--left c-flyout__content"
      ng-reflect-class="c-flyout__content--left"
      uiflyoutcontent=""
    >
      
      <ui-empty-state
        icon="users"
        ng-reflect-icon="users"
        ng-reflect-message="NO_USERS"
        ng-reflect-size="small"
        size="small"
      >
        <div
          class="c-empty-state c-empty-state--small"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-users"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            NO_USERS
          </p>
          <div
            ng-reflect-ng-class="[object Object]"
          />
        </div>
      </ui-empty-state>
      
    </div>
  </div>
</ui-user-select>
`;
